ecommerce/extensions/catalogue/migrations/0055_sf_opp_line_item_ent_attr.py

```
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+
+
+from django.db import migrations
+from oscar.core.loading import get_model
+
+from ecommerce.core.constants import COUPON_PRODUCT_CLASS_NAME
+
+ProductAttribute = get_model("catalogue", "ProductAttribute")
+ProductClass = get_model("catalogue", "ProductClass")
+
+SF_LINE_ITEM_ATTRIBUTE_CODE = 'salesforce_opportunity_line_item'
+
+
+def create_sf_opp_line_item_attribue(apps, schema_editor):
+    """Create enterprise coupon salesforce_opportunity_line_item product attribute."""
+    ProductAttribute.skip_history_when_saving = True
+    coupon = ProductClass.objects.get(name=COUPON_PRODUCT_CLASS_NAME)
```

[curious] Are there reasons to use `.get()` instead of `.get_object_or_404()` or a try/except?

---

`get_object_or_404()` really only applies in the context of a web request, not a DB migration.
Doing something like `except ProductClass.DoesNotExist` is technically safer, but we're operating in a context (Django migrations) where we're guaranteed that the item we're fetching will exist, because we know previous migrations have all executed.  We'd want the migration to fail hard if that object isn't there, because it means something has gone very wrong.